design-docs/handling-encodings.md

```diff
@@ -0,0 +1,87 @@
+# Encoding
+
+Updated: __2016/03/19__
+Status: __Under Review__
+
+This document describes the rules and conventions for handling the character encoding of file contents in general, and file names 
+and comments in archives. It applies to tasks which need to read, write and transform text files, as well as tasks which create
+archives (zip files, tar files, etc.)
+
+## Use Cases
+
+### File Content Encoding
+
+Any time we need to generate a text file, or to read, transform and write the contents of a text file, we need to know what 
+character encoding(s) to use.
+
+Common situations include:
+
+* Copying a file from one location to another.
+* Creating an archive (zip, tar, etc.).
+* Transforming text in a file.
+
+[GRADLE-1267](https://issues.gradle.org/browse/GRADLE-1267) describes just one instance of where this functionality is needed.
+
+### File Name Encoding
```

Comment:
The term "file name encoding" is a bit misleading. In reality, it's the encoding of entry names in archives that it is about in this section. And this can be specified in the ZipOutputStream.